tools/slo-rules-generator/slo-rules-generator.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"gopkg.in/yaml.v3"
+)
+
+
+const (
+	domainFileHeaderComment = `#
+# This file has been generated by slo-rules-generator.
+# DO NOT EDIT MANUALLY!
+#
+`
+)
+
+func yamlStr(s string) yaml.Node {
+	n := yaml.Node{}
+	n.SetString(s)
+	return n
+}
+
+type Labels map[string]string
+
+func (l Labels) Merge(with Labels) Labels {
+	out := Labels{}
+	for k, v := range l {
+		out[k] = v
+	}
+	for k, v := range with {
+		out[k] = v
+	}
+	return out
+}
+
+type SloConfiguration map[string]Domain
+
+func main() {
+	if len(os.Args) != 2 {
```

Comment:
Feature improvement:
Use [Kingpin](https://github.com/alecthomas/kingpin) which simply solves all of this